networkx/utils/backends.py

```diff
@@ -1,104 +1,172 @@
-environment variable ``NETWORKX_TEST_BACKEND`` is set to a registered
-backend key, the dispatch machinery will automatically convert regular
-networkx Graphs and DiGraphs to the backend equivalent by calling
-``<backend dispatcher>.convert_from_nx(G, edge_attrs=edge_attrs, name=name)``.
-Set ``NETWORKX_FALLBACK_TO_NX`` environment variable to have tests
-use networkx graphs for algorithms not implemented by the backend.
-
-The arguments to ``convert_from_nx`` are:
-
-- ``G`` : networkx Graph
-- ``edge_attrs`` : dict, optional
-    Dict that maps edge attributes to default values if missing in ``G``.
-    If None, then no edge attributes will be converted and default may be 1.
-- ``node_attrs``: dict, optional
-    Dict that maps node attribute to default values if missing in ``G``.
-    If None, then no node attributes will be converted.
-- ``preserve_edge_attrs`` : bool
-    Whether to preserve all edge attributes.
-- ``preserve_node_attrs`` : bool
-    Whether to preserve all node attributes.
-- ``preserve_graph_attrs`` : bool
-    Whether to preserve all graph attributes.
-- ``preserve_all_attrs`` : bool
-    Whether to preserve all graph, node, and edge attributes.
-- ``name`` : str
-    The name of the algorithm.
-- ``graph_name`` : str
-    The name of the graph argument being converted.
-
-The converted object is then passed to the backend implementation of
-the algorithm. The result is then passed to
-``<backend dispatcher>.convert_to_nx(result, name=name)`` to convert back
-to a form expected by the NetworkX tests.
-
-By defining ``convert_from_nx`` and ``convert_to_nx`` methods and setting
-the environment variable, NetworkX will automatically route tests on
-dispatchable algorithms to the backend, allowing the full networkx test
-suite to be run against the backend implementation.
-
-Example pytest invocation::
-
-    NETWORKX_TEST_BACKEND=sparse pytest --pyargs networkx
-
-Dispatchable algorithms which are not implemented by the backend
-will cause a ``pytest.xfail()``, giving some indication that not all
-tests are working, while avoiding causing an explicit failure.
-
-If a backend only partially implements some algorithms, it can define
-a ``can_run(name, args, kwargs)`` function that returns True or False
-indicating whether it can run the algorithm with the given arguments.
-
-A special ``on_start_tests(items)`` function may be defined by the backend.
-It will be called with the list of NetworkX tests discovered. Each item
-is a test object that can be marked as xfail if the backend does not support
-the test using `item.add_marker(pytest.mark.xfail(reason=...))`.
+NetworkX utilizes a plugin-dispatch architecture, which basically means we can
+easily plug in and out of backends with minimal code changes. Once a backend is
+connected through an entry_point to NetworkX, the flow of execution seamlessly
+dispatches (redirects) to the designated backend implementation, similar to how
```

Comment:
It might not hurt to also include a [link to the relevant section in the Python Packaging User Guide on entry points](https://packaging.python.org/en/latest/guides/creating-and-discovering-plugins/#using-package-metadata), either here or in the section below about creating a custom backend.